sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -182,6 +184,18 @@ namespace Azure { namespace Core {
       //  That implies we return T* and NOT a T&
     }
 
+    /**
+     * @brief Check whether the context has a deadline in it itself, or in the branch the context
+     * belongs to.
+     *
+     * @return `true` if this context, or the tree branch this context belongs to has a deadline
+     * associated with it. `false` otherwise.
+     */
+    bool HasDeadline() const
+    {
+      return static_cast<DateTime::time_point>(GetDeadline()) != (DateTime::max)();
+    }
```

Comment:
Can we defer adding this to later, once we see a need for it. @JeffreyRichter is this method really necessary for 1.0?

Reply:
No, we don't need this

Reply:
@Jinming-Hu asked for it. He would be using it: https://github.com/Azure/azure-sdk-for-cpp/pull/2010/files#diff-736005d9109cc9aa59c6ebc322fdb5dae5158a370c1ebb06ac7098a47561fc53R35-R40

What I can do, is to make an internal function.

Reply:
Just let @Jinming-Hu compare the deadline time with DateTime::max himself.
We dont' need functions for everything.

Reply:
ok

Reply:
@Jinming-Hu, `bool const hasDeadline = (ctx.GetDeadline() != Azure::DateTime::max());`

Reply:
I was thinking about this more. Why does @Jinming-Hu need this at all?
Most (maybe all) scenarios should just be caring whether the Context has reached deadline or not - the actual deadline time shouldn't matter.
If this is for passing timeout query param to service, then I think the policy should just add some timeout if the deadline is far enough into the future (say 1 week or something). Or, maybe do not add the timeout QP at all if the deadline is > 1 week out. 

I think that checking for the sentinel value is not the best way to go; someone could call WithDeadline(DateTime::max) to be explicit.